File: web/src/layout/LoginDemoXml.ts
//
// LoginDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './login_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.LoginDemoXml
export class LoginDemoXml {
    xmlRoot!: HTMLElement
    username!: HTMLInputElement
    password!: HTMLInputElement
    verifyPassword!: HTMLInputElement
    agree!: HTMLInputElement
    submitLoading!: HTMLDivElement
    submit!: HTMLButtonElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.username = getViewById<HTMLInputElement>(view, "username")
        this.password = getViewById<HTMLInputElement>(view, "password")
        this.verifyPassword = getViewById<HTMLInputElement>(view, "verifyPassword")
        this.agree = getViewById<HTMLInputElement>(view, "agree")
        this.submitLoading = getViewById<HTMLDivElement>(view, "submitLoading")
        this.submit = getViewById<HTMLButtonElement>(view, "submit")
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/LoginDemoXml.kt
//
// LoginDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class LoginDemoXml {

    lateinit var username: EditText
    lateinit var password: EditText
    lateinit var verifyPassword: EditText
    lateinit var agree: CheckBox
    lateinit var submitLoading: ViewFlipper
    lateinit var submit: Button
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.login_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        username = view.findViewById<EditText>(R.id.username)
        password = view.findViewById<EditText>(R.id.password)
        verifyPassword = view.findViewById<EditText>(R.id.verifyPassword)
        agree = view.findViewById<CheckBox>(R.id.agree)
        submitLoading = view.findViewById<ViewFlipper>(R.id.submitLoading)
        submit = view.findViewById<Button>(R.id.submit)
        
        
        
        return view
    }
}